src/Agent.Worker/ExecutionContext.cs

```diff
@@ -418,11 +419,12 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
                     Alias = "vsts_container_preview"
                 };
                 dockerContainer.Properties.Set("image", imageName);
-                Container = new ContainerInfo(HostContext, dockerContainer);
+                Container = HostContext.CreateContainerInfo(dockerContainer);
+                //Container = new ContainerInfo(HostContext, dockerContainer);
             }
             else if (!string.IsNullOrEmpty(message.JobContainer))
             {
-                Container = new ContainerInfo(HostContext, message.Resources.Containers.Single(x => string.Equals(x.Alias, message.JobContainer, StringComparison.OrdinalIgnoreCase)));
+                Container = new ContainerInfo(message.Resources.Containers.Single(x => string.Equals(x.Alias, message.JobContainer, StringComparison.OrdinalIgnoreCase)));
```

Comment:
this needs to be HostContext.CreateContainerInfo() as well

Reply:
Yup, modified it.